cmd/influx/task.go

```diff
@@ -3,342 +3,455 @@ package main
 
 	return cmd
 }
 
-func taskFindF(cmd *cobra.Command, args []string) error {
-	if err := taskFindFlags.org.validOrgFlags(&flags); err != nil {
+func (b *cmdTaskBuilder) taskFindF(cmd *cobra.Command, args []string) error {
+
+	if err := b.org.validOrgFlags(&flags); err != nil {
 		return err
 	}
 
-	client, err := newHTTPClient()
+	tskSvc, _, err := b.svcFn()
 	if err != nil {
 		return err
 	}
 
-	s := &http.TaskService{
-		Client: client,
-	}
-
 	filter := influxdb.TaskFilter{}
-	if taskFindFlags.user != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.user)
+	if b.taskFindFlags.user != "" {
+		id, err := influxdb.IDFromString(b.taskFindFlags.user)
 		if err != nil {
 			return err
 		}
 		filter.User = id
 	}
 
-	if taskFindFlags.org.name != "" {
-		filter.Organization = taskFindFlags.org.name
+	if b.org.name != "" {
+		filter.Organization = b.org.name
 	}
-	if taskFindFlags.org.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.org.id)
+	if b.org.id != "" {
+		id, err := influxdb.IDFromString(b.org.id)
 		if err != nil {
 			return err
 		}
 		filter.OrganizationID = id
 	}
 
-	if taskFindFlags.limit < 1 || taskFindFlags.limit > influxdb.TaskMaxPageSize {
+	if b.taskFindFlags.limit < 1 || b.taskFindFlags.limit > influxdb.TaskMaxPageSize {
 		return fmt.Errorf("limit must be between 1 and %d", influxdb.TaskMaxPageSize)
 	}
-	filter.Limit = taskFindFlags.limit
+	filter.Limit = b.taskFindFlags.limit
 
 	var tasks []*influxdb.Task
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if b.taskID != "" {
+		id, err := influxdb.IDFromString(b.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return b.printTasks(taskPrintOpts{tasks: tasks})
+}
+
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedCmd() *cobra.Command {
+	cmd := b.opts.newCmd("retry-failed", b.taskRetryFailedF, true)
+	cmd.Short = "Retry failed runs"
+	cmd.Aliases = []string{"rtf"}
+
+	b.org.register(b.opts.viper, cmd, false)
+	b.globalFlags.registerFlags(b.opts.viper, cmd)
+	registerPrintOptions(b.opts.viper, cmd, &b.taskPrintFlags.hideHeaders, &b.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&b.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.before, "before", "", "runs before this time")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.after, "after", "", "runs after this time")
+
```

Comment:
Can we add a boolean dry run flag? Or maybe better, a `--run` flag so the default is a dry run.